### PR TITLE
Override the default ssh askpass flow on project clone step

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
We support [project clone from ssh urls](https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#configuring-devworkspaces-to-use-ssh-keys-for-git-operations), but if the ssh key has a passphrase, the project clone step will fail because the default ssh askpass flow will be invoked, but it does not support any other input then a manual input from keyboard. To override the default flow, a script file is used. The passphrase file must be mounted to `/etc/ssh/` folder. The `SSH_ASKPASS` env variables defines the script file for the ssh flow, see: https://man.openbsd.org/ssh-add#ENVIRONMENT. The current `open-ssh` version does not support the `SSH_ASKPASS_REQUIRE` key, so we need to set the `DISPLAY` env variable as the command process does not have an associated terminal.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-6614
Fix https://github.com/devfile/devworkspace-operator/issues/1294

### Is it tested? How?

1. Setup ssh key with passphrase: https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#configuring-devworkspaces-to-use-ssh-keys-for-git-operations.
2. Try to start a workspace from an ssh url.
### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v8-devworkspace-operator-e2e, v8-che-happy-path` to trigger)
    - [ ] `v8-devworkspace-operator-e2e`: DevWorkspace e2e test
    - [ ] `v8-che-happy-path`: Happy path for verification integration with Che

### CODE REVIEW

**Comment by AObuchow** [comment]
Thanks for the PR @vinokurig  :) I didn't get a chance to properly look at this today, but I should be able to tomorrow (Thursday).

**Review by dkwon17 — APPROVED** [review]
I added my passphrase in the git-ssh-key secret, and it is working

**Review by AObuchow — COMMENTED** [review]
Left some minor comments regarding documentation but the functionality works as expected & I like this solution of leveraging ssh-add's features :)

File: docs/additional-configuration.adoc
Comment (vinokurig) [reply docs/additional-configuration.adoc]: fixed

File: docs/additional-configuration.adoc
Comment (vinokurig) [reply docs/additional-configuration.adoc:1]: done

File: docs/additional-configuration.adoc
Comment (vinokurig) [reply docs/additional-configuration.adoc:216]: I have tried the command without the `PASSPHRASE` env variable and it worked ok, but the `passphrase` key of the secret data has no value: 
```yaml
data:
  dwo_ssh_key: ***
  dwo_ssh_key.pub: ***
  passphrase: ''
  ssh_config: ***
```
Such empty keys will not be mounted to the container file system as files and if the `ssh-askpass.sh` is executed it will throw an error about missing passphrase file.

Comment (AObuchow) [reply docs/additional-configuration.adoc:216]: Thanks for checking. Just to confirm: if the user's SSH key does _not_ have a passphrase, will `ssh-askpass.sh` be invoked? I imagine it won't, based on it's documentation:

`If ssh-add **needs** a passphrase, it will read the passphrase from the current terminal if it was run from a terminal...`

The concern is just making sure that users are able to copy & paste the instructions and have them work, even if their SSH key does not have a passphrase (though their secret will have an empty passphrase data field).

Comment (vinokurig) [reply docs/additional-configuration.adoc:216]: > if the user's SSH key does not have a passphrase, will `ssh-askpass.sh` be invoked?

I have tried the flow with SSH key with an empty passphrase, and neither the default askpass scriptn nore the provided `ssh-askpass.sh` script was not invoked.

Comment (AObuchow) [reply docs/additional-configuration.adoc:216]: Perfect!

**Review by AObuchow — APPROVED** [review]
LGTM, great work @vinokurig :)
I've opened https://github.com/devfile/devworkspace-operator/issues/1294 for tracking in the DWO milestones.

Please squash your fixup commit before we merge this PR :)

**Comment by vinokurig** [comment]
@AObuchow Thank you for the review. Looks like we need @dmytro-ndp approval when he verifies the functionality.

File: project-clone/ssh-askpass.sh
Comment (ibuziuk) [inline project-clone/ssh-askpass.sh:5]: @vinokurig could you elaborate a bit? why do we exit 1 if the optional passphrase is missing?

**Review by ibuziuk — COMMENTED** [review]
@vinokurig have we checked that old ssh keys would be working as expected

**Comment by ibuziuk** [comment]
@musienko-maxim please review, we need QA approval for that one

File: project-clone/ssh-askpass.sh
Comment (vinokurig) [reply project-clone/ssh-askpass.sh:5]: Passphrase is optional because we do not need to specify passphrase if the ssh key is generated with an empty passphrase, but if the ssh key is generated **WITH** a passphrase we do not consider the passphrase as optional.

**Review by musienko-maxim — APPROVED** [review]

**Pull request merged**
